test/jdk/java/lang/reflect/code/TestRemoveFinalVars.java

```diff
@@ -0,0 +1,86 @@
+import org.testng.annotations.Test;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.code.Block;
+import java.lang.reflect.code.CopyContext;
+import java.lang.reflect.code.Op;
+import java.lang.reflect.code.Value;
+import java.lang.runtime.CodeReflection;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static java.lang.reflect.code.op.CoreOps.FuncOp;
+import static java.lang.reflect.code.op.CoreOps.VarAccessOp.VarLoadOp;
+import static java.lang.reflect.code.op.CoreOps.VarAccessOp.VarStoreOp;
+import static java.lang.reflect.code.op.CoreOps.VarOp;
+
+/*
+ * @test
+ * @run testng TestRemoveFinalVars
+ */
+
+public class TestRemoveFinalVars {
+
+    @CodeReflection
+    static int f(boolean b) {
+        final int x = 8; // final var
+        int y = x + 2; // final var
+        int z = y + 3; // non final var
+        z++;
+        return y;
+    }
+
+    @Test
+    void test() {
+        FuncOp f = getFuncOp(this.getClass(),"f");
+        f.writeTo(System.out);
+
+        FuncOp f2 = f.transform(TestRemoveFinalVars::rmFinalVars);
+        f2.writeTo(System.out);
+    }
+
+    /*
+    if VarOp
+        if varOp result not used with VarStore // final variable
+            do not copy the varOp
+            map varOp result with the initial value // a way to mark the var is final
+            do not copy VarLoads on the var and map their result with the init value
+
+    * */
+
+    static Block.Builder rmFinalVars(Block.Builder block, Op op) {
+        if (op instanceof VarOp varOp) {
+            if (isValueUsedWithOpClass(varOp.result(), VarStoreOp.class)) {
```

Comment:
Since this is an example we can improve this (since we might generalize on interfaces implementation by ops) by passing in a Predicate<Op> e.g., `op -> op instanceof VectorStoreOp` (then one can layer on top class equality as a helper function if needed).